src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -433,7 +435,13 @@ public function findAdjacentProgrammeByPosition(
             ->setParameter('parentDbId', $parentDbId)
             ->setParameter('originalPosition', $position);
 
-        return $qb->getQuery()->getOneOrNullResult(Query::HYDRATE_ARRAY);
+        $result = $qb->getQuery()->getOneOrNullResult(Query::HYDRATE_ARRAY);
+
+        if (!$result) {
+            return $result;
+        }
+        $this->addToAncestryCache([$result]);
```

Comment:
Please read the comment on that method. You don't have all the joins on findById, so this will cause problems.

Reply:
Best to remove it, not needed here anyway as we will already have the ancestors.